### PR TITLE
charCode shoud be limit to 16bit

### PR DESCRIPTION
**Pull request by sum2012**
Fix Juusei to Diamond chinese patch
translate.from ppsspp from https://github.com/hrydgard/ppsspp/pull/8135/

Before

After

### CODE REVIEW

**Comment by sum2012** [comment]
original log:
https://gist.github.com/sum2012/83a28943302b06fde803

modify log:
https://gist.github.com/sum2012/be30a92966c30e3f2ea1

File: src/jpcsp/HLE/modules/sceFont.java [hunk 1]
```diff
@@ -950,7 +950,7 @@ public int sceFontGetCharGlyphImage(int fontHandle, int charCode, TPointer glyph
                     buffer, bytesPerLine, bufWidth, bufHeight,
                     xPosI, yPosI, pixelFormat, charCode, font.fontLib.getAltCharCode());
         }
-
+        charCode&= 0xffff;
```
Comment (gid15) [inline src/jpcsp/HLE/modules/sceFont.java]: This should be done at the beginning of the function, not the end...

File: src/jpcsp/HLE/modules/sceFont.java [hunk 2]
```diff
@@ -1093,6 +1093,7 @@ public int sceFontGetCharGlyphImage_Clip(int fontHandle, int charCode, TPointer
                     buffer, bytesPerLine, bufWidth, bufHeight,
                     xPosI, yPosI, pixelFormat, charCode, font.fontLib.getAltCharCode());
         }
+        charCode&= 0xffff;
```
Comment (gid15) [inline src/jpcsp/HLE/modules/sceFont.java]: This should be done at the beginning of the function, not the end...

**Comment by sum2012** [comment]
done

**Comment by gid15** [comment]
Thank you!

**Comment by sum2012** [comment]
I  surprise that no game is known to use  unimplemented function to work correctly font

**Pull request merged**